paasta_tools/api/views/instance.py

```diff
@@ -240,21 +258,29 @@ def instance_status(request):
     instance_status['instance'] = instance
 
     try:
-        actual_deployments = get_actual_deployments(service, settings.soa_dir)
+        instance_type = validate_service_instance(service, instance, settings.cluster, settings.soa_dir)
     except Exception:
         error_message = traceback.format_exc()
         raise ApiFailure(error_message, 500)
 
-    version = get_deployment_version(actual_deployments, settings.cluster, instance)
-    # exit if the deployment key is not found
-    if not version:
-        error_message = 'deployment key %s not found' % '.'.join([settings.cluster, instance])
-        raise ApiFailure(error_message, 404)
+    if instance_type != 'flinkcluster':
```

Comment:
i'm bothered by growing amount of `if`s on instance_type. we  should refactor the whole thing into dynamic dispatch on  instance_type value, merging the result into `instance_state`, something like:

    INSTANCE_STATUS_HANDLERS = {'flinkcluster': ..., ...}
    def instance_status():
      instance_type = ...
      instance_status.update(INSTANCE_STATUS_HANDLERS[instance_type](service, instance, verbose, ...))
      return instance_status

Reply:
I'm totally agree with that, we need to do that.
Let's do this refactoring when we'll have a stable working code for several operators.